src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -462,75 +469,43 @@ private unsafe bool EndsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> suffi
         private const int FIND_FROMSTART = 0x00400000;
         private const int FIND_FROMEND = 0x00800000;
 
-        // TODO: Instead of this method could we just have upstack code call IndexOfOrdinal with ignoreCase = false?
-        private static unsafe int FastIndexOfString(string source, string target, int startIndex, int sourceCount, int targetCount, bool findLastIndex)
+        // TODO: Instead of this method could we just have upstack code call LastIndexOfOrdinal with ignoreCase = false?
+        private static unsafe int FastLastIndexOfString(string source, string target, int startIndex, int sourceCount, int targetCount)
         {
             int retValue = -1;
 
-            int sourceStartIndex = findLastIndex ? startIndex - sourceCount + 1 : startIndex;
+            int sourceStartIndex = startIndex - sourceCount + 1;
 
             fixed (char* pSource = source, spTarget = target)
             {
                 char* spSubSource = pSource + sourceStartIndex;
 
-                if (findLastIndex)
+                int startPattern = (sourceCount - 1) - targetCount + 1;
+                if (startPattern < 0)
+                    return -1;
+
+                char patternChar0 = spTarget[0];
```

Comment:
is target guaranteed to be non-empty? could we add an assert?

Reply:
Added